src/pal/src/misc/sysinfo.cpp

```diff
@@ -525,5 +525,20 @@ PAL_GetLogicalProcessorCacheSizeFromOS()
     }
 #endif
 
+#ifdef HAVE_SYSCTL
+    if (cacheSize == 0)
+    {
+        size_t cacheSizeFromSysctl = 0;
```

Comment:
The type of the value returned by the `hw.l3cachesize` is `int64_t`, not `size_t`. Only the `sz` is of size_t.